src/main/java/com/example/oddleweatherapi/controller/WeatherController.java

```diff
@@ -46,15 +48,22 @@ public void deleteWeather(@PathVariable("id") Integer id){
         weatherService.deleteWeather(id);
     }
 
-    @PutMapping(path = "{id}")
-    public void updateWeather(@PathVariable("id") Integer id,
-                              @RequestParam String temp,
-                              @RequestParam(required = false) String tempMin,
-                              @RequestParam(required = false) String tempMax,
-                              @RequestParam(required = false) String todayDate,
-                              @RequestParam Integer weatherTypeId,
-                              @RequestParam Integer cityId){
-        weatherService.updateWeather(id, temp, tempMin, tempMax, todayDate, weatherTypeId, cityId);
+//    @PutMapping(path = "{id}")
+//    public void updateWeather(@PathVariable("id") Integer id,
+//                              //request body weather dto
+//                              @RequestParam String temp,
+//                              @RequestParam(required = false) String tempMin,
+//                              @RequestParam(required = false) String tempMax,
+//                              @RequestParam(required = false) String todayDate,
+//                              @RequestParam Integer weatherTypeId,
+//                              @RequestParam Integer cityId){
+//        weatherService.updateWeather(id, temp, tempMin, tempMax, todayDate, weatherTypeId, cityId);
+//    }
+
+    @PutMapping
+    public void updateWeather(@RequestBody @Valid WeatherDTO weatherDTO){
```

Comment:
still need `path = "{id}"`

Reply:
Added path = "{id}" into Put Request and handle from id from request URL